### PR TITLE
Set progress when concatenating files.

### LINKED ISSUE #13610: No progress when concatenating downloaded segments for large download after transfer is complete
**Describe the bug**

I am using Backblaze S2, however I suspect that this issue will happen with all storage providers that support multipart downloads.   In my circumstance, I was downloading a 27GB file from Backblaze, which eventually hit 100% completion, however he transfers window still shows it as "Downloading" for several minutes while the speed drops slowly, before it says complete.   While it is stuck at 100% downloading, i can see the file in the downloads directory on my computer increasing in size until it hits the correct file size, at which point the transfers window shows the download as complete.   I suspect that as I have the "Segmented downloads with multiple connections per file" setting enabled, cyberduck is reconstructing the file, however the status on the transfers window doesn't change to indicate this, which could lead to confusion or people thinking downloads are not working correctly.

**To Reproduce**

1. Ensure Segmented download with multiple connections per file is enabled
2. Download a file tens of gigabytes or more in size
3. Wait until the download hits 100%

**Expected behavior**
If the issue is what i suspect it to be, i would expect the transfers window to change from Downloading 100% to something along the lines of either Reconstructing or Building InsertFileNameHere, which a percentage indicator showing its progress.

**Desktop (please complete the following information):**
Windows 11 21H2 (22000.795)

### PR DESCRIPTION
**Pull request by dkocher**
Fix #13610.

### CODE REVIEW

File: core/src/main/java/ch/cyberduck/core/transfer/download/AbstractDownloadFilter.java [hunk 1]
```diff
@@ -282,6 +282,8 @@ public void complete(final Path file, final Local local,
                 if(local.exists()) {
                     local.delete();
                 }
+                listener.message(MessageFormat.format(LocaleFactory.localizedString("Finalize {0}", "Status"),
```
Comment (AliveDevil) [inline core/src/main/java/ch/cyberduck/core/transfer/download/AbstractDownloadFilter.java:285]: Can we make this more userfriendly, and include ProgressReporting capabilities to the target-file WriteChannel?

**Comment by AliveDevil** [comment]
https://github.com/user-attachments/assets/b9ef481e-d433-4f04-a4e5-c95eaeeb03f2

Need to assess whether this behavior is tolerable.

**Comment by dkocher** [comment]
> Need to assess whether this behavior is tolerable.

It would be preferrable if the progress bar would change to indeterminate spinning while finalizing.

**Comment by AliveDevil** [comment]
So, while there is Progress in progress, disable Show/Open and do not rely on `Transfer.completed`?

**Review by AliveDevil — CHANGES_REQUESTED** [review]
Draft: Add Indeterminate state to macOS interface.

File: windows/src/main/csharp/ch/cyberduck/ui/ViewModels/TransfersViewModel.cs [hunk 2]
```diff
@@ -608,6 +615,7 @@ private void ValidateCommands()
         StopCommand.NotifyCanExecuteChanged();
         TrashCommand.NotifyCanExecuteChanged();
 
+        // TODO Reduce triggering this a billion times a second
```
Comment (dkocher) [inline windows/src/main/csharp/ch/cyberduck/ui/ViewModels/TransfersViewModel.cs:618]: @AliveDevil Resolve TODO

Comment (AliveDevil) [reply windows/src/main/csharp/ch/cyberduck/ui/ViewModels/TransfersViewModel.cs:618]: Not now. Would require reevaluating how all commands/notifications are handled in this Window. Future improvement.

**Review by ylangisc — APPROVED** [review]

**Pull request merged**